assets/docs/interactivity_tutorial.md

```diff
@@ -0,0 +1,441 @@
+
+Reactive rendering frameworks sometimes solve the problem of sharing mutable data via a shared local cache (store) that represents data from the server. LocusZoom can then make requests to that store, and the store is then responsible for deciding whether a new server request is needed. This allows the result of a single API request to power all widgets on the page without redundant network traffic. The store call can then return a promise representing either local data, or the result of a server request, as appropriate.
+
+In this model, if the data changed due to an external event, LocusZoom might need to be notified that new data is available. This is best done using the standard methods used to trigger a re-render from outside the plot  (`plot.applyState()`).
+
+If you are not using a store, this manual re-render can also be triggered by watching an observable value. (many frameworks provide a way to manually watch values for changes ([eg vue.js](https://vuejs.org/v2/guide/computed.html#Watchers)).
+
+## Filters control what is shown
+Filters can be used to control what elements are shown on a panel. This will hide elements, but preserve the space that those elements would have occupied: eg, the axis limits will reflect all of the data, not just what is currently shown.
+
+Filters are specified in data layout layouts, and can be applied to most data layer types.
+```javascript
+{
+  ...options,
+  filters: [
+    { field: '{{namespace[access]}}score', operator: '!=', value: null },
+  ],
+}
+```
+
+Scatter plots have an additional option to show labels for some or all data elements. This is controlled via adding a similar filter block inside *label*:
+```javascript
+{
+  id: 'phewaspvalues',
+  ...options,
+  label: {
+    text: '{{{{namespace[phewas]}}trait_label}}',
+    filters: [
+      {
+        field: '{{namespace[phewas]}}log_pvalue',
+        operator: '>=',
+        value: 20,
+      },
+    ],
+  }
+}
+```
+
+The following filters are available:
+
+* `=`: field value *exactly* matches the provided value
+* `!=`: field value *loosely* does not match the provided value (can be used to filter out falsy values such as undefined, null, or false)
+* Numerical comparisons: `<`, `<=`,  `>`, `>=`
+* `in`: field value is present inside the provided value. The provided value can be a string, or, more commonly, an array: eg "gene type is one of the following".
+* `match`: the provided value is present in the field value. Useful for partial text search, eg "show only genes whose name contains the user-provided filter string "HLA"
+
+> *NOTE:* If a list of multiple filters is provided, then *all* filter conditions must be true for an item to be shown in the plot.
+
+Filters can be modified interactively by the user after first render: see the `filter_field` widget for details.
+
+### Imperative API (provisional)
```

Comment:
TODO: deprecate and remove as of 207, and replace with filter registry for plugins.